### PR TITLE
JAMES-3320: Expose supported private extension in Session object

### PR DESCRIPTION
**Pull request by nvduc91**
When a user get a session object, urn:apache:james:params:jmap:mail:quota and urn:apache:james:params:jmap:mail:shares shoud appear next to urn:ietf:params:jmap:core and urn:ietf:params:jmap:mail

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala [hunk 1]
```diff
@@ -230,8 +230,15 @@ class Serializer @Inject() (mailboxIdFactory: MailboxId.Factory) {
       })
     }
 
-  private def mailboxWritesWithFilteredProperties(properties: Option[Properties], capabilities: Set[CapabilityIdentifier]): Writes[Mailbox] = {
-    mailboxWrites(Mailbox.propertiesFiltered(properties, capabilities))
+  private def mailboxWritesWithFilteredProperties(properties: Option[Properties], capabilities: immutable.Set[CapabilityIdentifier]): Writes[Mailbox] = {
```
Comment (remk) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala:233]: the `properties` parameter is not used anymore. Is it a  regression ?

Comment (remk) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala:233]: it seems that you ignored the change from JAMES-3098

Comment (remk) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala:233]: You should launch the `MailboxGetMethodContract` test

Comment (nvduc91) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala]: oh, i will recheck it.

**Comment by Arsnael** [comment]
Hmmm... what is supposed to do this PR? Not sure to get it...

**Comment by nvduc91** [comment]
> Hmmm... what is supposed to do this PR? Not sure to get it...

We should expose those two capabilities supported:
`urn:apache:james:params:jmap:mail:quota` and `urn:apache:james:params:jmap:mail:shares`

**Comment by Arsnael** [comment]
> why only the tests are modified? is there a missing file?

Was why I didn't get the point of the PR :)

**Comment by nvduc91** [comment]
> why only the tests are modified? is there a missing file?

Yes, im sorry, files missing.

> Was why I didn't get the point of the PR :)
You could said that at the first place, i so confused with your comment.

**Comment by nvduc91** [comment]
Failures: 
[3ae588382911865a4788ebe39f361e7c84a172ea] [ERROR]   RabbitMQEventDeadLettersIntegrationTest.multipleFailedEventShouldBeStoredInDeadLetter:359 1 expectation failed.
[3ae588382911865a4788ebe39f361e7c84a172ea] JSON path . doesn't match.
[3ae588382911865a4788ebe39f361e7c84a172ea] Expected: a collection with size <2>
[3ae588382911865a4788ebe39f361e7c84a172ea]   Actual: []

**Comment by nvduc91** [comment]
test this please

**Comment by chibenwa** [comment]
Part of https://github.com/linagora/james-project/pull/3594

**Pull request closed without merging**